workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -173,6 +173,12 @@ public synchronized WorkFlowExecution updateWorkFlow(WorkFlowExecution workFlowE
 		return workFlowRepository.save(workFlowExecution);
 	}
 
+	@Override
+	public List<WorkFlowStatusResponseDTO> getWorkFlowStatusByProjectId(UUID projectId) {
+		return workFlowRepository.findAllByProjectId(projectId).stream()
+				.map(workFlowExecution -> getWorkFlowStatus(workFlowExecution.getId())).toList();
```

Comment:
Why is the reason for this map?
Could you add a test for this?

Reply:
@eloycoto `stream().map()` allows applying a function to each stream element. So for me, it makes sense to iterate over the element and call `getWorkflowStatus` and then collect all results into a List. 
However, I agree that a unit test for this method is valuable.

Reply:
we have unit tests for `getWorkFlowStatus()`: https://github.com/RichardW98/parodos/blob/filter-workflow-by-projectid/workflow-service/src/test/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImplTest.java#L444